pkg/credentials/s3/utils.go

```diff
@@ -0,0 +1,105 @@
+package s3
+
+import v1 "k8s.io/api/core/v1"
+
+func BuildS3EnvVars(annotations map[string]string, s3Config *S3Config) []v1.EnvVar {
+	envs := []v1.EnvVar{}
+
+	if s3Endpoint, ok := annotations[InferenceServiceS3SecretEndpointAnnotation]; ok {
```

Comment:
this s3 endpoint if/else logic block is unchanged, just moved files from `s3_secret.go` to here since I use it in both `s3_secret.go` and `s3_service_account.go`